ci.js

```diff
@@ -0,0 +1,181 @@
+#! /usr/bin/env node
+
+const fs = require("fs").promises;
+const path = require("path");
+
+function printUsage() {
+    console.log(
+        `Usage: "ci check-consistency" or "ci bump-version %postfix%" or "ci get-version"`
+    );
+}
+
+let postfix;
+const mode = process.argv[2];
+
+switch (mode) {
+    case "get-version":
+        break;
+
+    case "bump-version":
+        postfix = process.argv[3];
+        if (!postfix) {
+            printUsage();
+            process.exit();
+        }
+        break;
+
+    case "":
+    case undefined:
+    case "check-consistency":
+        break;
+
+    default:
+        printUsage();
+        process.exit(0);
+}
+
+const pathToPackages = "./packages/";
+
+async function getFiles(currentPath, files) {
+    const entries = await fs.readdir(currentPath, { withFileTypes: true });
+
+    for (let file of entries) {
+        if (file.name === "node_modules") {
+            continue;
+        }
+
+        if (file.isDirectory()) {
+            await getFiles(`${currentPath}${file.name}/`, files);
+        } else if (file.name === "package.json") {
+            const packageJsonPath = path.join(
+                __dirname,
+                currentPath,
+                file.name
+            );
+            files.push(packageJsonPath);
+        }
+    }
+}
+
+async function doGetFiles(path) {
+    const files = [];
+    await getFiles(path, files);
+    return files;
+}
+
+async function getVersion(file) {
+    const content = await fs.readFile(file);
+    const json = JSON.parse(content);
+    return [json.name, json.version];
+}
+
+function processDep(obj, name, fn) {
+    if (!obj) {
+        return;
+    }
+
+    if (!obj[name]) {
+        return;
+    }
+
+    if (!/^workspace\:/.test(obj[name])) {
+        return;
+    }
+
+    const version = obj[name].replace("workspace:", "");
+    fn(obj, version);
+}
+
+async function getVersionsMap(allPackageJsons) {
+    const map = new Map();
+    for (let file of allPackageJsons) {
+        // console.log("Reading version from: ", file);
+        const [name, version] = await getVersion(file);
+        map.set(name, version);
+    }
+    return map;
+}
```

Comment:
here we can also avoid await in loop

const getVersionsMap = async (allPackageJsons) => new Map(
  await Promise.all(
    allPackageJsons.map((file) => getVersion(file))
  )
)